src/components/settings/GooglePermissions.js

```diff
@@ -0,0 +1,139 @@
+// @flow
+import React, { Component } from 'react';
+import GoogleLogin from 'react-google-login';
+import { connect } from 'react-redux';
+import { flashErrorMessage } from 'redux-flash';
+import type { AppState } from '../../state/AppState';
+import type { User } from '../../models/User';
+import {
+  connectGoogle, allowDriveAccess, refreshUser, revokeDriveAccess,
+} from '../../state/auth/auth.actionCreators';
+import googleDriveBadge from '../../assets/googleDrive.png';
+import './GooglePermissions.scss';
+
+type Props = {
+  user: User,
+  dispatch: Function,
+}
+
+type State = {
+  googleAuth: any,
+}
+
+class GooglePermissions extends Component<Props, State> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      googleAuth: null,
+    };
+  }
+
+  componentWillMount(): void {
+    const { dispatch, user } = this.props;
+    dispatch(refreshUser(user.id));
+    require('google-client-api')().then((gapi) => {
+      gapi.client.init({
+        clientId: process.env.REACT_APP_CLIENT_ID,
+        scope: 'profile email',
+      }).then(() => {
+        // Listen for sign-in state changes.
+        const googleAuth = gapi.auth2.getAuthInstance();
+        this.setState({ googleAuth });
+      });
+    });
+  }
+
+  isConnected = () => {
+    const { google } = this.props.user;
+    return !!google;
+  };
+
+  hasGoogleDriveAccess = () => {
+    const { google } = this.props.user;
+    if (google) {
+      return google.hasOfflineAccess;
+    }
+    return false;
+  };
+
+  handleConnectGoogleSuccess = (response) => {
+    const { user, dispatch } = this.props;
+    dispatch(connectGoogle(user.id, response.getAuthResponse().id_token));
+  };
+
+  handleConnectGoogleFailure = () => {
+    const { dispatch } = this.props;
+    dispatch(flashErrorMessage('Connect to Google failed'));
+  };
+
+  requestAdditionalPermissions = () => {
+    const { googleAuth } = this.state;
+    googleAuth.grantOfflineAccess(
+      { scope: 'profile email https://www.googleapis.com/auth/drive' },
+    ).then(this.additionalPermissionsSuccess, this.additionalPermissionsError);
+  };
+
+  additionalPermissionsSuccess = (googleResponse) => {
+    const { user, dispatch } = this.props;
+    dispatch(allowDriveAccess(user.id, googleResponse.code));
+  };
+
+  additionalPermissionsError = () => {
+    this.props.dispatch(flashErrorMessage('Google Drive permissions declined'));
```

Comment:
Please be consistent with deconstructing. See this method in line [64](https://github.com/hpi-sam/ask-your-repository-web/pull/64/files#diff-300ae6b621a733b5db3618ad91df8643R64).